### ISSUE TITLE
Importing an isolated vertex or edge (non-manifold) into Blender with GoB makes it invisible even in edit mode and cannot be selectable.

### ISSUE THREAD

**Issue by Tsukasa-YAMATO** [JoseConseco/GoB#104]
This maybe  a bug of GoB or Blender's side.

environment
Windows 10 Pro
Blender 2.83
GoB 3.3.7
Zbrush 2020.1.4

-----------------------------
Before I tried GoB Ver.3.3.8 beta, I have found a bug, so now I report it. (I will try it later with Ver.3.3.8, but it will probably be reproduced.)

The reproduction procedure is shown below.

1. Open the Dog project in Zbrush and send to Blender with GoB
2. Enter edit mode on Blender
3. Select one vertex, press SHIFT+D and move it

![2020-06-11_01h36_35](https://user-images.githubusercontent.com/65609416/84294978-ec7eff00-ab84-11ea-9816-e88cdd5b555b.jpg)

4. Export GoB to Zbrush
5. Import into Blender again with GoB
6. Enter edit mode and press A key
7. Not all vertices are in the selected state even if you press the A key (see image)

![2020-06-11_01h46_25](https://user-images.githubusercontent.com/65609416/84295657-d7ef3680-ab85-11ea-9730-c2e4407d0183.jpg)

The isolated vertex exists as data ( I confirmed it by Zbrush).

![2020-06-11_01h44_42](https://user-images.githubusercontent.com/65609416/84295211-42ec3d80-ab85-11ea-9293-2c94748c0306.jpg)

But  in Blender's edit mode it's invisible, and cannot even be selected. It's very weird.

In the same procedure as above, duplicate one edge instead of the vertex with SHIFT+D and isolated it as a non-manifold. After exporting to Zbrush and import it again, it appears as a orange colored bar in object mode, but it also becomes invisible in edit mode again (the same as the vertex's test case).

![2020-06-11_01h53_48](https://user-images.githubusercontent.com/65609416/84296148-96ab5680-ab86-11ea-9dd5-194508b6bb4c.jpg)

This isolated edge is not also selectable, and the two vertices (the edge itself) remain unselected when I use the A key to select all.

-------------

(Found Solution but not essentially)

To find a solution, I duplicated and deleted the vertex and edge of the mesh in question with SHIFT+D. And using 3D-Print Toolbox. 

![2020-06-11_02h01_26](https://user-images.githubusercontent.com/65609416/84296929-b68f4a00-ab87-11ea-8861-15cfca81f1d0.jpg)

 In the process, the invisible vertex and edge became visible. It's strange. It seems important to use 3D-Print Toolbox and checking it. (But the finding is just by chance.)

And once they become visible, they can be detected properly even by checking non-manifolds.

![2020-06-11_02h04_35](https://user-images.githubusercontent.com/65609416/84297092-f5250480-ab87-11ea-86fe-9d9cf1d7d190.jpg)

This may be a bug related to Face Maps transfer. (Because neither vertex nor edge has Face.)

And as I expected, isolated Faces don't have this problem, I tested it. It may also be related to Face Maps.

------------

My preferences  settings are below.

![2020-06-11_01h01_58](https://user-images.githubusercontent.com/65609416/84290845-395fd700-ab7f-11ea-893f-3e7746995c42.jpg)

**Comment by Tsukasa-YAMATO** [comment]
@kromar 

I just came up with it, so I'll comment before I forget it.
I came up with something might be UV related. Because I remembered this bug and your comment.

https://github.com/JoseConseco/GoB/issues/95#issuecomment-633926320

(Of course, it's possible that it's irrelevant. I'm sorry if so.)

I hope you find it helpful.

**Comment by kromar** [comment]
This is a strange one, i am able to reproduce it and spent quite some time trying to understand the problem, im not sure yet why this happen. 
Its a strange issue that should not happen and im not sure if this might be a bug in blender. I will do some more testing and see if i can fix this.

**Comment by Tsukasa-YAMATO** [comment]
@kromar 
Dear Daniel,
Thank you for confirming. I am pleased that you have reproduced it.
Single isolated vertex and edge in Blender display correctly in edit mode. In particular, the A key is displayed correctly.
However, it becomes invisible via the GoB addon. If I consider only this, GoB may have a bug.

But maybe there's a problem with the Python provided by Blender, as you pointed out. It's beyond my knowledge, so I have no idea.

After investigating it for a period of time, and if it's not clear, describe it as a "known bug" and put this issue on hold. You should not have better to delay the release of stable version regardless of this issue. Because this has nothing to do with transferring the weighted mesh between Zbrush and Blender. It's just a matter of mesh integrity.

Many users are more interested in "weighted mesh transfer" than such a trivial problem.
And the solution is simple. Users can use Zbrush, Blender's addon or any other tool, and they can fix it by feature of fixing the integrity of the mesh.

You have tarrent and may try to solve this problem. However, I think it is courageous to give up once if you can't solve it after some period of trial and error. (And the possibility of a bug on the Blender's side can't be denied. With that in mind, it may not be a good idea to stick to solve.)

(I have deleted the word "fatal" from the first post.)

**Comment by kromar** [comment]
@Tsukasa-YAMATO i made a fix for this issue and also there was also something that could cause blender to crash on import so that should also be fixed.
One limitation however that comes from the GOZ format is that it has no edge information, so if you send a mesh for the first time from zbrush to blender that has "edges without faces only the vertices can be built in blender. Not much i can do about that limitation.
![image](https://user-images.githubusercontent.com/1472884/84443365-bd799380-ac3f-11ea-8169-b5cae872f1ff.png)


https://github.com/kromar/blender_GoB/tree/bugfix/facemapsexport

**Comment by Tsukasa-YAMATO** [comment]
@kromar 
Good morning, Mr. Daniel. Japan is 7 o'clock but I'm still in bed. After 12 hours or later, I will check the content of comment and  the new GoB version.

Anyway thank you for your Struggle.
Please give me some time.
Zzzz

**Comment by Tsukasa-YAMATO** [comment]
@kromar 
Thank you for your quick fix.
I confirmed what you were saying about the issue of isolated non-manifolds.

Isolated vertices or edges are now correctly selected in Blender's edit mode, even through GoB. So it will be found them when checking non-manifolds. There is no problem anymore.

Sending an edge-only non-manifold mesh from Zbrush to Blender "for the first time" is a reduction to only isolated vertices in edit mode, but that's hardly a problem. 
Because uses can not generate "mesh with only edges without faces" in Zbrush via general operations. (I have no idea how to generating it even if I use ZModeler, essentially though no need for knowing that :) )


So I've confirmed what you mean by exporting it in Obj format with Blender, importing it into Zbrush and then sending it to Blender with GoB at first time. However, as long as users do not do such a special thing, this phenomenon does not occur.
As long as users handle " a mesh that does not contain non-manifold " with Zbrush and sculpt it, there is no problem.

Anyway, and essentially non-manifolds should be removed, so if it is correctly detected and removed by Blender and Zbrush's mesh checker, then there is no problem.

(Also, I understand that there is no countermeasure for the file format limit. So I think this is enough to deal with this bug.)


Thank you for your fix.
I will continue the beta test and I feel this is getting to the completed thing more and more.
